src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -1133,12 +1133,33 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             me->GetMotionMaster()->MovePoint(SMART_RANDOM_POINT, x, y, z);
             break;
         }
-        case SMART_ACTION_RISE_UP:
+        case SMART_ACTION_MOVE_OFFSET:
         {
             if (!me)
                 break;
 
-            me->GetMotionMaster()->MovePoint(SMART_RANDOM_POINT, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ() + (float)e.action.moveRandom.distance);
+            ObjectList* targets = GetTargets(e, unit);
+            if (targets)
+            {
+                float x, y, z, o;
+                for (ObjectList::const_iterator itr = targets->begin(); itr != targets->end(); ++itr)
+                {
```

Comment:
And isnt there a pre check for certain smart_actions to not put in other objects than creatures into? I think I saw something in SmartScriptMgr or so...